bot/seasons/evergreen/trivia_quiz.py

```diff
@@ -139,36 +149,56 @@ def check(m: discord.Message) -> bool:
                     else:
                         await ctx.send(f"Cmon guys, {30-hint_no*10}s left!")
 
+                # Once hint or time alerts has been sent 2 times, the hint_no value will be 3
+                # If hint_no > 2, then it means that all hints/time alerts have been sent.
+                # Also means that the answer is not yet given and the bot sends the answer and the next question.
                 else:
                     response = random.choice(WRONG_ANS_RESPONSE)
                     expression = random.choice(ANNOYED_EXPRESSIONS)
                     await ctx.send(f"{response} {expression}")
                     await self.send_answer(ctx.channel, question_dict)
                     await asyncio.sleep(1)
-                    hint_no = 0
-                    unanswered += 1
-                    await self.send_score(ctx.channel, self.player_dict[ctx.channel.id])
+
+                    hint_no = 0  # init hint_no = 0 so that 2 hints/time alerts can be sent for the new question.
+
+                    await self.send_score(ctx.channel, self.game_player_scores[ctx.channel.id])
                     await asyncio.sleep(2)
 
             else:
+                # Reduce points by 25 for every hint/time alert that has been sent.
                 points = 100 - 25*hint_no
-                if msg.author in self.player_dict[ctx.channel.id]:
-                    self.player_dict[ctx.channel.id][msg.author] += points
+                if msg.author in self.game_player_scores[ctx.channel.id]:
+                    self.game_player_scores[ctx.channel.id][msg.author] += points
+                else:
+                    self.game_player_scores[ctx.channel.id][msg.author] = points
+
+                # Also updating the overall scoreboard.
+                if msg.author in self.player_scores:
+                    self.player_scores[msg.author] += points
                 else:
-                    self.player_dict[ctx.channel.id][msg.author] = points
+                    self.player_scores[msg.author] = points
+
                 hint_no = 0
-                unanswered = 0
+
                 await ctx.send(f"{msg.author.mention} got the correct answer :tada: {points} points for ya.")
                 await self.send_answer(ctx.channel, question_dict)
-                await self.send_score(ctx.channel, self.player_dict[ctx.channel.id])
+                await self.send_score(ctx.channel, self.game_player_scores[ctx.channel.id])
                 await asyncio.sleep(2)
 
+    @commands.command(name="scoreboard")
```

Comment:
The scoreboard command shouldn't be a top level command, but in a `quiz` command group.

I am also not sure if the command should be called `scoreboard`, as it is actually a bot session leaderboard. The scoreboard would be the current ranking of players in a current game. `.quiz leaderboard` would be a better name.